tokio/src/fs/write.rs

```diff
@@ -24,6 +24,49 @@ use std::{io, path::Path};
 /// # }
 /// ```
 pub async fn write(path: impl AsRef<Path>, contents: impl AsRef<[u8]>) -> io::Result<()> {
+    #[cfg(all(tokio_uring, feature = "rt", feature = "fs", target_os = "linux"))]
+    {
+        let handle = crate::runtime::Handle::current();
+        let driver_handle = handle.inner.driver().io();
+        if driver_handle.check_and_init()? {
+            return write_uring(path, contents.as_ref()).await;
+        }
+    }
+
+    write_spawn_blocking(path, contents).await
+}
+
+#[cfg(all(tokio_uring, feature = "rt", feature = "fs", target_os = "linux"))]
+async fn write_uring(path: impl AsRef<Path>, mut buf: &[u8]) -> io::Result<()> {
```

Comment:
I would change both methods to take `&Path` and `crate::util::as_ref::OwnedBuf`, to avoid generics in the actual implementations.